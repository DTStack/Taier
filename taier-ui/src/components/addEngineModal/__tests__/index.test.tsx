import { cleanup, fireEvent, render, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import AddEngineModal from '..';

describe('Test AddEngineModal Component', () => {
    beforeEach(() => {
        cleanup();
        document.body.innerHTML = '';
    });
    it('Should match snapshot', async () => {
        const { asFragment } = render(<AddEngineModal title="新增集群" visible />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Should validate form before confirm', async () => {
        const fn = jest.fn();
        const { getByTestId, container, getByText } = render(<AddEngineModal title="新增集群" visible onOk={fn} />);

        fireEvent.click(getByTestId('antd-mock-Modal-confirm'));
        await waitFor(() => {
            expect(getByText('集群标识不可为空！')).toBeInTheDocument();
        });

        fireEvent.change(container.querySelector('#clusterName')!, {
            target: { value: 'abc-abc' },
        });

        await waitFor(() => {
            expect(getByText('集群标识不能超过64字符，支持英文、数字、下划线')).toBeInTheDocument();
        });

        fireEvent.change(container.querySelector('#clusterName')!, {
            target: { value: new Array(100).fill('a').join('') },
        });

        await waitFor(() => {
            expect(getByText('集群标识不能超过64字符，支持英文、数字、下划线')).toBeInTheDocument();
        });

        fireEvent.change(container.querySelector('#clusterName')!, {
            target: { value: 'abc' },
        });
        fireEvent.click(getByTestId('antd-mock-Modal-confirm'));

        await waitFor(() => {
            expect(fn).toBeCalledWith({ clusterName: 'abc' });
        });
    });
});
