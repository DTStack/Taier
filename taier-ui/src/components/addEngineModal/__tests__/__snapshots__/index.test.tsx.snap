// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddEngineModal Component Should match snapshot 1`] = `
<DocumentFragment>
  <p
    data-testid="antd-mock-Modal-title"
  >
    新增集群
  </p>
  <form
    autocomplete="off"
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
        >
          <label
            class="ant-form-item-required"
            for="clusterName"
            title="集群名称"
          >
            集群名称
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-required="true"
                class="ant-input"
                id="clusterName"
                placeholder="请输入集群标识"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
  <button
    data-testid="antd-mock-Modal-confirm"
  >
    confirm
  </button>
</DocumentFragment>
`;
